tests/python/pants_test/init/test_plugin_resolver.py

```diff
@@ -3,222 +3,269 @@
+from pants.engine.fs import CreateDigest, Digest, DirectoryToMaterialize, FileContent, MergeDigests
+from pants.engine.process import Process, ProcessResult
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
 from pants.init.plugin_resolver import PluginResolver
 from pants.option.options_bootstrapper import OptionsBootstrapper
-from pants.python.setup_py_runner import SetupPyRunner
 from pants.testutil.interpreter_selection_utils import (
     PY_36,
     PY_37,
     python_interpreter_path,
     skip_unless_python36_and_python37_present,
 )
-from pants.testutil.subsystem.util import init_subsystem
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.test_base import TestBase
 from pants.util.contextutil import temporary_dir
-from pants.util.dirutil import safe_rmtree, touch
+from pants.util.dirutil import safe_mkdir, safe_rmtree, touch
 
-req = Requirement.parse
+DEFAULT_VERSION = "0.0.0"
 
 
-class Installer(metaclass=ABCMeta):
-    def __init__(self, source_dir: Path, install_dir: Path) -> None:
-        self._source_dir = source_dir
-        self._install_dir = install_dir
+class PluginResolverTest(TestBase):
+    @classmethod
+    def rules(cls):
+        return (
+            *super().rules(),
+            *pex.rules(),
+            *download_pex_bin.rules(),
+            *python_native_code.rules(),
+            *subprocess_environment.rules(),
+            *external_tool.rules(),
+            *archive.rules(),
+            RootRule(PexRequest),
+        )
 
-    def run(self) -> None:
-        init_subsystem(SetupPyRunner.Factory)
-        dist = self._create_dist(SetupPyRunner.Factory.create())
-        shutil.copy(str(dist), str(self._install_dir))
+    def _create_pex(self) -> Pex:
+        return self.request_single_product(
+            Pex,
+            Params(
+                PexRequest(
+                    output_filename="setup-py-runner.pex",
+                    requirements=PexRequirements(["setuptools==44.0.0", "wheel==0.34.2"]),
+                ),
+                create_options_bootstrapper(args=["--backend-packages=pants.backend.python"]),
+                SubprocessEncodingEnvironment(None, None),
+                PexBuildEnvironment(tuple(), tuple()),
+            ),
+        )
 
-    @abstractmethod
-    def _create_dist(self, runner: SetupPyRunner) -> Path:
-        ...
+    def _run_setup_py(
+        self, plugin: str, version: str, setup_py_args: Iterable[str], install_dir: str
+    ) -> None:
+        pex_obj = self._create_pex()
+        source_digest = self.request_single_product(
+            Digest,
+            CreateDigest(
+                [
+                    FileContent(
+                        "setup.py",
+                        dedent(
+                            f"""
+                    from setuptools import setup
+
+                    setup(name="{plugin}", version="{version or DEFAULT_VERSION}")
+                """
```

Comment:
Nit: Black is horrible at formatting multiline strings. Ideally, the strings would be indented by 4 more spaces and the closing """ by 8 spaces.